### DISCUSSION TITLE
Compile schism on TACC frontera

### DISCUSSION THREAD

**Discussion by yunfangsun** [noaa-ocs-modeling/CoastalApp-testsuite#9] [General]
With the module load impi-intel/esmf-8.1.1 and ./build.sh --compiler intel --component="schism" -tp parmetis -plat tacc

I get the following errors:

Install the project...
/usr/bin/cmake -P cmake_install.cmake
-- Install configuration: ""
-- Installing: /work2/09283/yfsun001/frontera/new1/CoastalApp/thirdparty_open/parmetis/del/include/parmetis.h
-- Installing: /work2/09283/yfsun001/frontera/new1/CoastalApp/thirdparty_open/parmetis/del/lib/libparmetis.a
-- Installing: /work2/09283/yfsun001/frontera/new1/CoastalApp/thirdparty_open/parmetis/del/bin/ptest
-- Installing: /work2/09283/yfsun001/frontera/new1/CoastalApp/thirdparty_open/parmetis/del/bin/mtest
-- Installing: /work2/09283/yfsun001/frontera/new1/CoastalApp/thirdparty_open/parmetis/del/bin/parmetis
-- Installing: /work2/09283/yfsun001/frontera/new1/CoastalApp/thirdparty_open/parmetis/del/bin/pometis
make[1]: Leaving directory `/work2/09283/yfsun001/frontera/new1/CoastalApp/thirdparty_open/parmetis/build/Linux-x86_64'
Removing: /work2/09283/yfsun001/frontera/new1/CoastalApp/THIRDPARTY_INSTALL/lib/libmetis.a
Removing: /work2/09283/yfsun001/frontera/new1/CoastalApp/THIRDPARTY_INSTALL/lib/libparmetis.a
WARN:: deleteDIR (compileMetis:main): called from: functions_utilities (functions_build:build.sh)
        deleted the directory: /work2/09283/yfsun001/frontera/new1/CoastalApp/thirdparty_open/parmetis/del

WARN:: compileMetis (main): called from: functions_build (build.sh)
        Setting the following variables:
           PARMETISHOME = /work2/09283/yfsun001/frontera/new1/CoastalApp/THIRDPARTY_INSTALL
           METISHOME    = /work2/09283/yfsun001/frontera/new1/CoastalApp/THIRDPARTY_INSTALL
           METIS        = /work2/09283/yfsun001/frontera/new1/CoastalApp/THIRDPARTY_INSTALL
           METIS_PATH   = /work2/09283/yfsun001/frontera/new1/CoastalApp/THIRDPARTY_INSTALL

compileNems :: Compiling: make -j 1 -f GNUmakefile build COMPONENTS="SCHISM"
NOTE: Skipping appbuilder.mk creation; no appbuilder file in use.
echo 'SCHISM' > "/work2/09283/yfsun001/frontera/new1/CoastalApp/NEMS/src/conf/components.mk"
NEMS_BUILDOPT IS
cp /work2/09283/yfsun001/frontera/new1/CoastalApp/conf/configure.nems.linux.intel /work2/09283/yfsun001/frontera/new1/CoastalApp/NEMS/src/conf/configure.nems
cp /work2/09283/yfsun001/frontera/new1/CoastalApp/conf/externals.nems /work2/09283/yfsun001/frontera/new1/CoastalApp/NEMS/src/conf/externals.nems
cat /dev/null > /work2/09283/yfsun001/frontera/new1/CoastalApp/NEMS/src/conf/modules.nems
cp /work2/09283/yfsun001/frontera/new1/CoastalApp/NEMS/src/ESMFVersionDefine_ESMF_NUOPC.h /work2/09283/yfsun001/frontera/new1/CoastalApp/NEMS/src/ESMFVersionDefine.h
( echo '. /work2/09283/yfsun001/frontera/new1/CoastalApp/NEMS/src/conf/modules.nems' ) > "/work2/09283/yfsun001/frontera/new1/CoastalApp/NEMS/src/conf/modules.nems.sh"
( echo 'source /work2/09283/yfsun001/frontera/new1/CoastalApp/NEMS/src/conf/modules.nems' ) > "/work2/09283/yfsun001/frontera/new1/CoastalApp/NEMS/src/conf/modules.nems.csh"
/work2/09283/yfsun001/frontera/new1/CoastalApp/NEMS/src/incmake/configure_rules.mk:3: /work2/09283/yfsun001/frontera/new1/CoastalApp/NEMS/src/conf/test-results.mk: No such file or directory
make -f /work2/09283/yfsun001/frontera/new1/CoastalApp/NEMS/src/incmake/tests.mk    \
      MODULE_LOGIC="echo No module selected."           \
      TARGET="/work2/09283/yfsun001/frontera/new1/CoastalApp/NEMS/src/conf/test-results.mk" TEST
make[1]: Entering directory `/work2/09283/yfsun001/frontera/new1/CoastalApp/NEMS'
if [ -f "/modules.nems" ] ; then \
  echo No module selected. ; \
fi ; \
( \
echo "# Do not edit this file.  It is automatically generated from tests.mk." ; \
echo "CASELESS_FILESYSTEM=NO" ; \
if [ "Q$ESMFMKFILE" != Q ] ; then egrep 'VERSION|^ESMF_[A-Z]*DIR' < "$ESMFMKFILE" ; fi ; \
) > "/work2/09283/yfsun001/frontera/new1/CoastalApp/NEMS/src/conf/test-results.mk"
make[1]: Leaving directory `/work2/09283/yfsun001/frontera/new1/CoastalApp/NEMS'
echo No module selected.; echo "SCHISM_SRCDIR = /work2/09283/yfsun001/frontera/new1/CoastalApp/SCHISM/schism/src"; exec cmake -S /work2/09283/yfsun001/frontera/new1/CoastalApp/SCHISM/schism/src -B /work2/09283/yfsun001/frontera/new1/CoastalApp/SCHISM/build -DCMAKE_VERBOSE_MAKEFILE=TRUE \
-DCMAKE_Fortran_COMPILER=mpiifort -DCMAKE_CXX_COMPILER=mpiicpc -DCMAKE_C_COMPILER=mpiicc -DOLDIO=ON -DUSE_WW3=ON -DPREC_EVAP=OFF
No module selected.
SCHISM_SRCDIR = /work2/09283/yfsun001/frontera/new1/CoastalApp/SCHISM/schism/src
Re-run cmake no build system arguments
CMake Error: The source directory "/work2/09283/yfsun001/frontera/new1/CoastalApp/SCHISM/build" does not exist.
Specify --help for usage, or press the help button on the CMake GUI.
make: *** [/work2/09283/yfsun001/frontera/new1/CoastalApp/SCHISM_INSTALL/schism.mk] Error 1

Could you please help?

**Comment by pvelissariou1** [comment]
Resolved with the installation of the new libraries and module system. Tested with FVCOM, SCHISM and ADCIRC test cases.
